### ISSUE TITLE
Unable to import extensions in Python 3.12 / ModuleNotFoundError

### ISSUE THREAD

**Issue by brentvollebregt** [bottlepy/bottle#1430]
After upgrading to Python 3.12, importing an extension, for example `from bottle.ext.websocket import websocket`, now gives the following error:

```
Traceback (most recent call last):
  File "C:\...\test.py", line 1, in <module>
    from bottle.ext.websocket import websocket
ModuleNotFoundError: No module named 'bottle.ext.websocket'
```

**My environment:**

- Windows 10
- Python 3.12.0
- bottle 0.12.25
- bottle-websocket 0.2.9 (the example extension)

**Reproduction:**

1. Install dependencies: `pip install bottle==0.12.25 bottle-websocket==0.2.9`
2. Run script:
    ```py
    from bottle.ext.websocket import websocket
    ```

**Investigation / Potential Solution:**

I see this error does not occur on master but it does on the release-0.12 branch / 0.12.25 tag. Using the changes from https://github.com/bottlepy/bottle/commit/ca6762c559c5e71e0dff71dc97eb4c6b3ed9bbcd that are currently on master seems to fix this issue.

**Comment by brentvollebregt** [comment]
To add more info to this, I see support for the `find_module` API has been removed in Python 3.12 as [noted in the changelog here](https://docs.python.org/3/whatsnew/3.12.html#importlib):

> Support for `find_loader()` and `find_module()` APIs have been removed. (Contributed by Barry Warsaw in [gh-98040](https://github.com/python/cpython/issues/98040).)

The changelog recommends using `find_spec()` instead.

**Comment by defnull** [comment]
This is already implemented in 0.13-dev and will be released soon.
